analytic_base_department/models/analytic.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2011-2016 Camptocamp SA
+# Copyright 2013 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    department_id = fields.Many2one(
+        'hr.department',
+        'Department')
+
+
+class AnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.model
+    def _default_department(self):
+        department_id = False
+        employee = self.env.user.employee_ids[0]
```

Comment:
This line is not ok. In case `self.env.user.employee_ids` is empty, you get this:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/home/odoo/OCB-11.0/addons/hr_timesheet/models/hr_timesheet.py", line 34, in create
    return super(AccountAnalyticLine, self).create(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 3255, in create
    vals = self._add_missing_default_values(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 1528, in _add_missing_default_values
    defaults = self.default_get(list(missing_defaults))
  File "/home/odoo/OCB-11.0/addons/hr_timesheet/models/hr_timesheet.py", line 12, in default_get
    result = super(AccountAnalyticLine, self).default_get(field_list)
  File "/.repo_requirements/odoo/odoo/models.py", line 1009, in default_get
    defaults[name] = field.default(self)
  File "/home/odoo/build/OCA/account-analytic/analytic_base_department/models/analytic.py", line 29, in <lambda>
    default=lambda self: self._default_department(),
  File "/home/odoo/build/OCA/account-analytic/analytic_base_department/models/analytic.py", line 21, in _default_department
    employee = self.env.user.employee_ids[0]
  File "/.repo_requirements/odoo/odoo/models.py", line 4661, in __getitem__
    return self._browse((self._ids[key],), self.env)
IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 928, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-39>", line 2, in button_immediate_install
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 448, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/.repo_requirements/odoo/odoo/addons/base/module/module.py", line 541, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/.repo_requirements/odoo/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 343, in load_modules
    loaded_modules, update_module)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 242, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 159, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='demo')
  File "/.repo_requirements/odoo/odoo/modules/loading.py", line 94, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 788, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 849, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 748, in parse
    exc_info[2]
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/.repo_requirements/odoo/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/home/odoo/OCB-11.0/addons/hr_timesheet/models/hr_timesheet.py", line 34, in create
    return super(AccountAnalyticLine, self).create(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 3255, in create
    vals = self._add_missing_default_values(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 1528, in _add_missing_default_values
    defaults = self.default_get(list(missing_defaults))
  File "/home/odoo/OCB-11.0/addons/hr_timesheet/models/hr_timesheet.py", line 12, in default_get
    result = super(AccountAnalyticLine, self).default_get(field_list)
  File "/.repo_requirements/odoo/odoo/models.py", line 1009, in default_get
    defaults[name] = field.default(self)
  File "/home/odoo/build/OCA/account-analytic/analytic_base_department/models/analytic.py", line 29, in <lambda>
    default=lambda self: self._default_department(),
  File "/home/odoo/build/OCA/account-analytic/analytic_base_department/models/analytic.py", line 21, in _default_department
    employee = self.env.user.employee_ids[0]
  File "/.repo_requirements/odoo/odoo/models.py", line 4661, in __getitem__
    return self._browse((self._ids[key],), self.env)
odoo.tools.convert.ParseError: "tuple index out of range" while parsing /home/odoo/OCB-11.0/addons/hr_timesheet/data/hr_timesheet_demo.xml:4, near
<record id="working_hours_requirements" model="account.analytic.line">
        <field name="name">Requirements analysis and specification</field>
        <field name="user_id" ref="base.user_root"/>
        <field name="date" eval="(DateTime.now() + relativedelta(days=-1)).strftime('%Y-%m-%d')"/>
        <field name="unit_amount">2.00</field>
        <field name="project_id" ref="project.project_project_2"/>
        <field name="amount">-60.00</field>
    </record>
```

Steps to reproduce in Runbot:

- set the field user_id of the Administrator employee to False (empty field)
- try to install the timesheet app

Instead you could write something like this:

```
    employee = self.env.user.employee_ids
    if employee and employee[0].department_id:
        department_id = employee[0].department_id.id
```

It would be very nice if you can add a test to verify that `def _default_department()` works fine in such case.